src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1669,8 +1676,15 @@ impl<'a> StringReader<'a> {
     fn scan_raw_byte_string(&mut self) -> token::Lit {
         let start_bpos = self.pos;
         self.bump();
-        let mut hash_count = 0;
+        let mut hash_count: u8 = 0;
         while self.ch_is('#') {
+            if hash_count == 255 {
+                let last_bpos = self.pos;
+                self.fatal_span_(start_bpos,
+                                    last_bpos,
+                                    "too many `#` characters surrounding a raw string; \
```

Comment:
Maybe clarify to "raw byte string". It's probably fine to only test one of them, though, as it's a minor error. Also, the indentation here is off.